src/OrleansAzureUtils/Providers/Streams/AzureQueue/AzureQueueAdapterReceiver.cs

```diff
@@ -85,7 +84,7 @@ public Task Initialize(TimeSpan timeout)
                 IEnumerable<CloudQueueMessage> messages = await task;
                 
                 List<IBatchContainer> azureQueueMessages = messages
-                    .Select(msg => (IBatchContainer)AzureQueueBatchContainer.FromCloudQueueMessage(msg, lastReadMessage++)).ToList();
```

Comment:
lastReadMessage++ orders events by the order they were read from queue and preserves that order in the cache.  Cache may not function correctly if read order does not match ordering info in message.